google/cloud/storage/well_known_parameters.h

```diff
@@ -431,6 +431,14 @@ struct Prefix : public internal::WellKnownParameter<Prefix, std::string> {
   static char const* well_known_parameter_name() { return "prefix"; }
 };
 
+/**
+ *
```

Comment:
Missing documentation?

Reply:
Doh! Fixed.

Reply:
Can you document how a prefix distinguished from a leaf object in ObjectMetadata?

Reply:
There is no difference.  GCS really does not have directories, as the documentation says:

https://cloud.google.com/storage/docs/cloud-console#_creatingfolders

> Because the Cloud Storage system **has no notion of folders**, folders created in the Cloud Console are a convenience to help you organize objects in a bucket. 

You can create an object called "foo/bar/baz" and there may or may not be any objects called  called "foo/" and/or "foo/bar/".  Or you can create an object *with text content* called "foo/" and another with text content called "foo/bar".

gsutil uses some conventions that may be relevant for your use-case:

https://cloud.google.com/storage/docs/gsutil/addlhelp/HowSubdirectoriesWork